packages/at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -429,23 +443,45 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
   }
 
   /// Syncs the cloud secondary changes to local secondary.
-  Future<List<KeyInfo>> _syncFromServer(
-      int serverCommitId,
-      int lastReceivedServerCommitId,
-      List<CommitEntry> uncommittedEntries) async {
+  Future<List<KeyInfo>> _syncFromServer(int serverCommitId,
+      int lastReceivedServerCommitId, List<CommitEntry> uncommittedEntries,
+      {int? localCommitIdBeforeSync}) async {
     // Iterates until serverCommitId is greater than lastReceivedServerCommitId.
     // replacing localCommitId with lastReceivedServerCommitId fixes infinite loop issue
     // in certain scenarios e.g server has a commit entry that need not be synced on client side,
     // server has delete commit entry and the key is not present on local keystore
     List<KeyInfo> keyInfoList = [];
     try {
+      int? skipDeletesUntil;
+      AtValue? isInitialSyncDone;
+      try {
+        isInitialSyncDone = await _atClient.get(_isInitialSyncDone);
+      } on AtKeyNotFoundException {
+        //ignore
+      }
+      // Set skipDeletesUntil when local commit id is -1 and
+      // isInitialSyncDone is not set
+      if (localCommitIdBeforeSync == -1 && isInitialSyncDone == null) {
```

Comment:
@VJag 
This check will not work when initial sync is interrupted.
if i try, localCommitIdBeforeSync == -1 || isInitialSyncDone == null then this will run for apps which have already run initial sync